engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -167,17 +167,19 @@ public ValidatorImpl(ConstraintValidatorFactory constraintValidatorFactory,
 	public final <T> Set<ConstraintViolation<T>> validate(T object, Class<?>... groups) {
 		Contracts.assertNotNull( object, MESSAGES.validatedObjectMustNotBeNull() );
 
-		if ( !beanMetaDataManager.isConstrained( object.getClass() ) ) {
```

Comment:
Mmmh, but in fact, we can get rid of these too. I got too conservative and wanted to keep the `ValidationContext` initialization after the `hasConstraints` check but it's a bad idea.

I changed it and got rid of `isConstrained()`.